src/data/bundle.lean

```diff
@@ -0,0 +1,60 @@
+/-
+Copyright © 2021 Nicolò Cavalleri. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nicolò Cavalleri
+-/
+
+import tactic.basic
+import algebra.module.basic
+
+/-!
+# Bundle
+Basic data structure to implement fiber bundles, vector bundles (maybe fibrations?), etc. This file
+should contain all possible results that do not involve any topology.
+We provide a type synonym of `Σ x, E x` as `bundle.total_space E`, to be able to endow it with
+a topology which is not the disjoint union topology. In general, the constructions of fiber bundles
+we will make will be of this form.
+## References
+- https://en.wikipedia.org/wiki/Bundle_(mathematics)
+-/
+
+namespace bundle
+
+variables {B : Type*} (E : B → Type*)
+
+/--
+`total_space E` is the total space of the bundle `Σ x, E x`. This type synonym is used to avoid
+conflicts with general sigma types.
+-/
+def total_space := Σ x, E x
+
+instance [inhabited B] [inhabited (E (default B))] :
+  inhabited (total_space E) := ⟨⟨default B, default (E (default B))⟩⟩
+
+/-- `bundle.proj E` is the canonical projection `total_space E → B` on the base space. -/
+@[simp] def proj : total_space E → B := sigma.fst
+
+instance {x : B} : has_coe_t (E x) (total_space E) := ⟨λ y, (⟨x, y⟩ : total_space E)⟩
```

Comment:
Does this work?
```suggestion
instance {x : B} : has_coe_t (E x) (total_space E) := ⟨sigma.mk x⟩
```

Reply:
Yes is it better? I find how it is now more explicative both in the definition and in proofs.

Reply:
It's better here because it doesn't introduce a lambda. Your other lemmas are fine as they are.

Reply:
Why is introducing lambdas a bad thing?